server/plugin/api.go

```diff
@@ -1258,6 +1422,297 @@ func (p *Plugin) getRepositories(c *UserContext, w http.ResponseWriter, r *http.
+	if issue.Milestone > 0 {
+		ghIssue.Milestone = &issue.Milestone
+	}
+
+	currentUser, appErr := p.API.GetUser(c.UserID)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load current user", StatusCode: http.StatusInternalServerError})
+		return
+	}
+
+	splittedRepo := strings.Split(issue.Repo, "/")
+	owner := splittedRepo[0]
+	repoName := splittedRepo[1]
+
+	githubClient := p.githubConnectUser(c.Context.Ctx, c.GHInfo)
+
+	result, resp, err := githubClient.Issues.Edit(c.Ctx, owner, repoName, issue.IssueNumber, ghIssue)
+	if err != nil {
+		if resp != nil && resp.Response.StatusCode == http.StatusGone {
+			p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "Issues are disabled on this repository.", StatusCode: http.StatusMethodNotAllowed})
+			return
+		}
+		c.Log.WithError(err).Warnf("Failed to update issue")
+		p.writeAPIError(w,
+			&APIErrorResponse{
+				ID: "",
+				Message: "failed to update issue: " + getFailReason(resp.StatusCode,
+					issue.Repo,
+					currentUser.Username,
+				),
+				StatusCode: resp.StatusCode,
+			})
+		return
+	}
+
+	rootID := issue.PostID
+	channelID := issue.ChannelID
+	message := fmt.Sprintf("Updated GitHub issue [#%v](%v)", result.GetNumber(), result.GetHTMLURL())
+	if post != nil {
+		if post.RootId != "" {
+			rootID = post.RootId
+		}
+		channelID = post.ChannelId
+		message += fmt.Sprintf(" from a [message](%s)", permalink)
+	}
+
+	reply := &model.Post{
+		Message:   message,
+		ChannelId: channelID,
+		RootId:    rootID,
+		UserId:    c.UserID,
+	}
+
+	if post != nil {
+		_, appErr = p.API.CreatePost(reply)
+	} else {
+		p.API.SendEphemeralPost(c.UserID, reply)
+	}
+	if appErr != nil {
+		c.Log.WithError(appErr).Warnf("failed to create notification post")
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to create notification post, postID: " + issue.PostID + ", channelID: " + channelID, StatusCode: http.StatusInternalServerError})
+		return
+	}
+
+	post, appErr = p.API.GetPost(issue.PostID)
+	if appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + issue.PostID, StatusCode: http.StatusInternalServerError})
+		return
+	}
+	if post == nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post " + issue.PostID + ": not found", StatusCode: http.StatusNotFound})
+		return
+	}
+	post.Props["assignees"] = issue.Assignees
+	post.Props["labels"] = issue.Labels
+	post.Props["description"] = issue.Body
+	post.Props["title"] = issue.Title
+	if _, appErr = p.API.UpdatePost(post); appErr != nil {
+		p.writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to update post " + issue.PostID, StatusCode: http.StatusInternalServerError})
+	}
+
+	p.writeJSON(w, result)
+}
+
+func (p *Plugin) CreateCommentToIssue(c *UserContext, w http.ResponseWriter, comment string, owner string, repo string, post *model.Post, numberInt int) {
```

Comment:
Better naming than numberInt
```suggestion
func (p *Plugin) CreateCommentToIssue(c *UserContext, w http.ResponseWriter, comment, owner, repo string, post *model.Post, numberInt int) {
```